### PR TITLE
implement proxy->delete() and use it in tests

### PR DESCRIPTION
**Pull request by davidlmorton**

### CODE REVIEW

**Comment by davidlmorton** [comment]
I've confirmed by hand that all workflows created by running implementation tests are deleted.

File: integration_tests/cancel_basic_workflow.t [hunk 1]
```diff
@@ -24,6 +24,12 @@ $wf_proxy->cancel;
 $wf_proxy->wait(polling_interval => 1);
 is($wf_proxy->status, 'canceled', 'Got expected (canceled) status');
 is_deeply($wf_proxy->outputs, undef, 'Got expected (empty) outputs');
+
+# this keeps us from deleting the workflow before petri can send
+# 'set_dag_status_running' callbacks
```
Comment (mkiwala) [inline integration_tests/cancel_basic_workflow.t]: Why is it important to not delete the workflow before petri can send the 'set_dag_status_running' callbacks?

Comment (davidlmorton) [inline integration_tests/cancel_basic_workflow.t]: It's just noise, all the tests have already finished by this time.  Additionally, it returns 500 which is retried by the petri service making it noise that persists.

**Comment by mkiwala** [comment]
+1

**Pull request merged**